### ISSUE TITLE
V2 breaks signInWithPassword on node (desktop app, not a web server)

### ISSUE THREAD

**Issue by irreal** [supabase/supabase-js#571]
# Bug report

## Describe the bug

V2 currently breaks our usage from a nodejs desktop application. This is not a web server that uses the private token in admin mode, nor does it "impersonate" a client using setAuthheader. It's a standalone desktop application that uses auth.signInWithPassword();

The sign in attempt is succesfull, but none of the db related methods seem to utilize the session. Only the realtime module "sees" the session.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a completely empty node js app
2. Install Supabase JS V2 (RC10 at the time of writing)
3. create an index.js file and inside do:

```
const { createClient } = require("@supabase/supabase-js");

const client = createClient("INSERT_HOST_HERE", "INSERT_PUBLIC_ANON_KEY_HERE");

client.auth.onAuthStateChange((event, session) => {
  if (event === "SIGNED_IN") {
    client
      .from("table")
      .select("*")
      .then(({ data, error }) => {
        //if RLS is checking email or uid in the policy, this will not get the rows that match
      });
    client
      .channel("*")
      .on("postgres_changes", { event: "*", schema: "*" }, (payload) => {
        //this will work just fine, it passes the session to the api and gets notifications according to RLS
      })
      .subscribe();
  }
});

client.auth.signInWithPassword({email: 'insert valid email here', password: 'insert valid password here'});
```

## Expected behavior

The db api call should use the session that was just established

## System information

- OS: macOS
- Version of supabase-js: V2 RC 10
- Version of Node.js: v18.9.0

## Additional context

Since the exact same code works just fine in the browser, wIthout going deep into the sdk code, this looks like an issue with localStoarge not being available in node?
However, I patched in localStorage and it still could not get the established session. No difference at all.

I know this usecase isn't very common, but it works awesomely for us on V1.x and I'm worried we will get stuck on it forever.
If you have any pointers, I could take a stab at a fix.

Thanks so much for such an awesome open source project! ❤️

**Comment by soedirgo** [comment]
Hey! I actually bumped into this recently - can you try setting [`persistSession: false`](https://supabase.com/docs/reference/javascript/next/initializing#parameters)? This only happens on Node afaik - we'll add a note on this on the reference docs.

**Comment by irreal** [comment]
Awesome, that totally works. Setting persistSession to false makes everything work again. I understand that it will not persist the session when the app is closed and then launched again later, but this the behaviour we want, so that's cool.

**Comment by joedalsin** [comment]
We ran into this issue as well writing jest tests for our RLS rules.  We use the locally served anon key and sign in with auth.signInWithPassword().   In supabase-js 1.x the signIn method worked fine.  With 2.x the switch to signInWithPassword() mysteriously started failing by failing all RLS rules.   Glad we found this AND it would be great if there were sample of E2E test for RLS rules so that devs can write and test locally.
